dataloader/dataloader.hpp

```diff
@@ -250,6 +380,82 @@ class DataLoader
     return index;
   }
 
+  /**
+   * Performs train test split.
+   *
+   * @tparam DatasetType Type of input dataset.
+   * @tparam LabelsType Type of input labels.
+   *
+   * @param dataset Features of dataset.
+   * @param labels Labels of the dataset.
+   * @param validRatio Ratio for train-test split.
+   * @param shuffle Boolean to determine shuffling of dataset.
+   */
+  template<typename TrainY = DatasetY,
+      typename std::enable_if_t<std::is_same<arma::field<arma::vec>, TrainY>, int> = 0>
+  void TrainTestSplit(DatasetX dataset,
```

Comment:
Hey @KimSangYeon-DGU, I have implemented the split for field type as well. It works if we isolate it from the rest of the dataloader. The issue is that with the dataloader it gives a compilation error i.e. if DatasetY = arma::mat then we are trying to insert vector instead of double. So I tried various combinations of enable_if to have two functions, One that would work for matrix type and other for field type. However I haven't yet been able to resolve the issue. The other PR works for matrix with data split and everything but I think field type is the way to go. Kindly let me know what you think and any suggestions to resolve the issue.

Reply:
I think I figured something. Let me try it out.

Reply:
This works mostly as work around, Let me know if this makes sense and I'll complete the rest. One more reason to use field is that in split function an image may end up in training and validation both and might have different bounding boxes. Here each image only has its corresponding bounding boxes. And now we can support both mat type and field type.

Reply:
@kartikdutt18 Nice, let's work with this PR instead of #13 because this approach gives us more flexible way.

Reply:
Great, I'll add support for matrix type shortly.